DMCompiler/DM/Expressions/Constant.cs

```diff
@@ -173,6 +177,17 @@ class Number : Constant {
             return new Number(Location, Value % rhsNum.Value);
         }
 
+        public override Constant ModuloModulo(Constant rhs) {
+            if (rhs is not Number rhsNum) {
+                return base.Add(rhs);
```

Comment:
This probably means to call `base.ModuloMudulo()`

..So does every other math operation override that seems to have copy-pasted `base.Add()`
```suggestion
                return base.ModuloModulo(rhs);
```